rust/src/jsonbuilder.rs

```diff
@@ -651,38 +705,62 @@ impl JsonBuilder {
                     &buf[0..offset],
                     val.as_bytes(),
                 );
-                self.buf.push_str(&error);
+                self.push_str(&error)?;
             }
         }
         Ok(())
     }
+
+    fn encode_base64(&mut self, val: &[u8]) -> Result<&mut Self, JsonError> {
+        let encoded_len = ((val.len() / 4) * 3) + 2;
```

Comment:
This looks wrong : if val.len() == 1, this gives 2 instead of 4

This should rather be `4*((val.len()+2)/3)`, no ?

Reply:
Fixed.